### PR TITLE
Adding events to itinerary

### LINKED ISSUE #56: Implement full itinerary page functionality
**User Story**
As a user, I want to see the itinerary that has already been created for my road trip, so that I can see what we are doing on our trip.

**Acceptance Criteria**

- Hook up frontend to backend
- Persist any data in the database
- Organiser can add items to itinerary
- Style and implement final page with itinerary displayed

**Additional Context**
Screen Shot 2021-04-27 at 4 30 49 PM

**Dependencies**
N/A

### PR DESCRIPTION
**Pull request by Jennak21**
The user is now able to add events to the days of their road trip
image

Please note styling and sizing is a bit jank. If the description is one long word, excess will be hidden.
If the screen size is too small a scrollbar appears for the event to show the rest of notes and descriptions.
image

Also I think there is an issue with airbnb react-dates component. If the input date changes and has a different month, the calendar will not automatically switch to that month. 

Closes #56

### CODE REVIEW

File: frontend/src/components/Itinerary/DailyItinerary/DayCard/AddEventModal/AddEventModal.jsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            }}
+            inputProps={{
+              step: 600, // 10 min
+            }}
+            onChange={(event) => setTime(event.target.value)}
+          />
+          <TextField defaultValue="" fullWidth label="Notes" onChange={(event) => setNotes(event.target.value)} />
+        </DialogContent>
+        <DialogActions>
+          <Button onClick={handleClose} color="primary">
```
[the remark this answers is not in the document]
Comment (Denijar) [reply frontend/src/components/Itinerary/DailyItinerary/DayCard/AddEventModal/AddEventModal.jsx:43]: actually same for the modal with the datepicker

File: frontend/src/components/Itinerary/ItineraryWrapper/ItineraryWrapper.jsx [hunk 2]
```diff
@@ -11,5 +11,5 @@ export default function ItineraryWrapper() {
   const { response, loading } = useGet(URL);
 
   // show spinner while data is being fetched
-  return <div>{loading || !response ? <Spinner /> : <Itinerary itineraryData={response.data} />}</div>;
+  return <>{loading || !response ? <Spinner /> : <Itinerary itineraryData={response.data} />}</>;
```
Comment (Denijar) [inline frontend/src/components/Itinerary/ItineraryWrapper/ItineraryWrapper.jsx:14]: currently the React file structure here breaks convention with the ItineraryWrapper folder being inside the Itinerary folder, even though the Wrapper is the outer component. Can this be refactored?

File: frontend/src/components/Itinerary/DailyItinerary/DayCard/AddEventModal/AddEventModal.jsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    addEvent(event);
+    handleClose();
+  };
+
+  return (
+    <div>
+      <Dialog fullWidth open={open} onClose={handleClose}>
+        <DialogTitle>What would like to do?</DialogTitle>
+        <DialogContent>
+          <TextField fullWidth autoFocus label="Description" onChange={(event) => setDescription(event.target.value)} />
```
Comment (Denijar) [inline frontend/src/components/Itinerary/DailyItinerary/DayCard/AddEventModal/AddEventModal.jsx:26]: nit: I think having about 10px margin between the text fields would be nice

**Review by Denijar — APPROVED** [review]
JENNNAAA this PR pops off I love the way you have implemented this it's so good. Some hella tiny comments but man this is sick. I really don't think the styling jank is worth worrying about at this point.

**Review by Denijar — CHANGES_REQUESTED** [review]
Also question - so we are bringing back a "card" style for this page, which I think makes sense for the nature of this page. Maybe for consistency should I put back the shadow around the PackingList? What do you think?

**Comment by lhuijser** [comment]
> Also question - so we are bringing back a "card" style for this page, which I think makes sense for the nature of this page. Maybe for consistency should I put back the shadow around the PackingList? What do you think?

Yeah it makes sense for the nature of this page- now that this is here I think it makes sense if you want to add the shadow back to the PackingList, but only if you can be bothered, I think it's good either way!

**Review by lhuijser — APPROVED** [review]
This is CLEAN AS HECK Jenna you are a star!! Thank you for this fab implementation!! Love it

**Review by Denijar — APPROVED** [review]
Nice thanks for that file structure change. Very happy to approve that suggestion about the margins was an absolute nit so alggg

**Review by courtneyeh — APPROVED** [review]
Looks amazing!! Amazing job :)

**Pull request merged**